project/apps/web/app/questions/components/EditModal.tsx

```diff
@@ -0,0 +1,218 @@
+                <FormItem>
+                  <FormLabel className="text-black">Title</FormLabel>
+                  <FormControl>
+                    <Input placeholder="Enter title" {...field} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Description Field */}
+            <FormField
+              control={form.control}
+              name="q_desc"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel className="text-black">Description</FormLabel>
+                  <FormControl>
+                    <Textarea
+                      className="resize-y max-h-64"
+                      placeholder="Enter description"
+                      spellCheck={false}
+                      {...field}
+                    />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Complexity Dropdown */}
+            <FormField
+              control={form.control}
+              name="q_complexity"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel className="text-black">Complexity</FormLabel>
+                  <FormControl>
+                    <Select value={field.value} onValueChange={field.onChange}>
+                      <SelectTrigger>
+                        <SelectValue placeholder="Select complexity" />
+                      </SelectTrigger>
+                      <SelectContent>
+                        {Object.values(COMPLEXITY).map((level) => (
+                          <SelectItem key={level} value={level}>
+                            {level}
+                          </SelectItem>
+                        ))}
+                      </SelectContent>
+                    </Select>
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Categories Multi-Select */}
+            <FormField
+              control={form.control}
+              name="q_category"
+              render={() => (
+                <FormItem>
+                  <FormLabel className="text-black">Categories</FormLabel>
+                  <FormControl>
+                    <div className="flex flex-wrap gap-2">
+                      {categories.map((cat) => (
+                        <Badge
+                          key={cat}
+                          onClick={() => toggleCategory(cat)}
+                          variant={
+                            selectedCategories.includes(cat)
+                              ? "default"
+                              : "secondary"
+                          }
+                          className="cursor-pointer"
+                        >
+                          {cat}
+                        </Badge>
+                      ))}
```

Comment:
same as above suggestion too. also we can consider using multi-select dropdown if number of categories becomes too long. something like [this](https://shadcn-multi-select-component.vercel.app/)

Reply:
Tried integrating this multi-select dropdown, but the scrollbar doesn't work on wheel.

Will stick to the current implementation for now, and change if there are too many categories in the future.